src/vm/assembly.cpp

```diff
@@ -216,8 +216,9 @@ void Assembly::Init(AllocMemTracker *pamTracker, LoaderAllocator *pLoaderAllocat
     // methods as intrinsics.
     LPCUTF8 assemblyName = GetSimpleName();
     const int length = sizeof("System.Numerics") - 1;
-    if ((strncmp(assemblyName, "System.Numerics", length) == 0) &&
-        ((assemblyName[length] == '\0') || (strcmp(assemblyName+length, ".Vectors") == 0)))
+    if (((strncmp(assemblyName, "System.Numerics", length) == 0) &&
+        ((assemblyName[length] == '\0') || (strcmp(assemblyName+length, ".Vectors") == 0))) ||
+        IsSystem())
```

Comment:
I expect that this will have significant impact on JIT throughout - the JIT will treat every type in CoreLib as potential SIMD intrinsic now.

Reply:
Yes, I guess so. Is there any possible solution?

Reply:
What are the operations that you need?

I see one: Given intrinsic argument, find out whether it is Vector128/Vector256 and the generic instantiation details.

What other operations you need?

Reply:
> I see one: Given intrinsic argument, find out whether it is Vector128/Vector256 and the generic instantiation details.

And the return values from intrinsics and `Unsafe.Read`?

Reply:
Why do you need to recognize `Unsafe.Read`?

Reply:
Hmm.. maybe we do not need. I just worry about the `__vectorcall` and System V calling convention, which we will implement in the future.

Reply:
Searching mscorlib is not a feasible solution. I assume that the only methods of interest would have generic arguments that are SIMD types (otherwise they couldn't be defined in mscorlib). I *believe* it is the case that, when you encounter a such a method, the JIT will already have seen SIMD types (from the type of the call or the type of the argument), in which case `compiler->usesSIMDTypes()` will return true. So, you could expand the search for a SIMD intrinsic to search non-SIMD-assembly methods that have SIMD types as generic arguments, and do that only if `usesSIMDTypes()` returns true, to avoid penalizing methods that don't use SIMD.

Reply:
> I assume that the only methods of interest would have generic arguments that are SIMD types (otherwise they couldn't be defined in mscorlib). 

@CarolEidt Thank you so much for reminding the generic problem that I neglected😅. I will try.

Additionally, how can I avoid search mscorlib for instantiated SIMD types?

Reply:
> how can I avoid searching mscorlib for instantiated SIMD types?

What specific intrinsics are you trying to recognize?
I don't think you need/want to involve the vm. If you want to recognize an mscorlib method as an intrinsic, you would probably want to change `Compiler::impSIMDIntrinsic()` to check for the case you're interested in, e.g. not just checking for `isSIMDClass(clsHnd)`, but also checking whatever you're looking for (e.g. return type).

> I just worry about the __vectorcall and System V calling convention, which we will implement in the future.

I think that's a completely orthogonal issue. When we get to that point, it will be up to the JIT to ensure that the calling convention is applied consistently across managed code.

Reply:
I had not realized that these types are actually declared in mscorlib. That makes them fundamentally different than the other SIMD types. For mscorlib intrinsics, we now have named intrinsics, but I don't believe that mechanism was intended to support a large number of methods, and there's no mechanism for adding a new type. I think some design work is needed to determine how to support this effectively (i.e. without impacting JIT throughput).

Reply:
> That makes them fundamentally different than the other SIMD types.
> I think some design work is needed to determine how to support this effectively (i.e. without impacting JIT throughput).

Yes, that is what I attempted to ask. 
For JIT throughput with named intrinsics, as we discussed previously, we can try to optimize the table search by binary search or hashtable. For the new SIMD type recognition, can we directly expose the SIMD class handles from vm rather than search them in JIT?